package/upgrade/upgrade_manifests.sh

```diff
@@ -526,54 +598,345 @@ loop_wait_rollout_event() {
+  if [ $enabled = true ]; then
+    echo "enable logging addon"
+    yq -i e '.spec.enabled = true' $target_file
+  else
+    echo "disable logging addon"
+    yq -i e '.spec.enabled = false' $target_file
+  fi
+
+  echo "final logging addon yaml file"
+  cat $target_file
+
+  kubectl apply -f $target_file
+  rm -f $target_file
+  rm -f $saved_config_file
+  kubectl get addon.harvesterhci.io -A || echo "fail to get addon"
+
+  if [ $enabled = true ]; then
+    # wait for addon to be applied
+    echo "wait for rollout of logging and audit"
+    sleep 30
     # loop wait for at most 6 minutes (36 * 10s)
     loop_wait_rollout_logging_audit 36
 
-    # due to error: unable to recognize "./rancher-logging.yaml": no matches for kind "EventTailer" in version "logging-extensions.banzaicloud.io/v1alpha1"
-    # the eventtailer needs to be deployed after the managedcharts are deployed
+    pod_cnt=$(kubectl get pods -n "$namespace" --no-headers | wc -l) || echo "fail to get pods in $namespace"
+    echo "there are $pod_cnt pods in $namespace after upgrade"
+  fi
 
-    # prepare a malformed yaml file, make sure it is effectively replaced
-    echo "to-be-replaced-file" > rancher-logging.yaml
+  echo "Logging Event Audit: finish converting"
+}
 
-    # reuse frame work to generate yaml file
-    # rancher-logging_event-extension is reusing chart rancher-logging, but as an extension for event
-    upgrade_managed_chart_from_version $UPGRADE_PREVIOUS_VERSION rancher-logging_event-extension rancher-logging.yaml
+convert_monitoring_to_addon() {
+  echo "Monitoring: start to convert monitoring from managedchart to addon"
+  local chart_name="rancher-monitoring"
+  local namespace="cattle-monitoring-system"
+  local cnt=0
+
+  # check if monitoring addon is there
+  cnt=$(kubectl get addon.harvesterhci.io -n "$namespace" "$chart_name" --no-headers | wc -l)
+  if [ "$cnt" -gt 0 ]; then
+    echo "monitoring has already been addon"
+    return 0
+  fi
 
-    echo "Apply resource file of event"
+  local src_file=/usr/local/share/addons/monitoring_addon.yaml
+  if [ ! -f $src_file ]; then
+    echo "there is no $src_file, cannot convert"
+    return 0
+  fi
 
-    kubectl apply -f ./rancher-logging.yaml
+  # check if monitoring managedchart is there
+  cnt=$(kubectl get managedchart -n fleet-local "$chart_name" --no-headers | wc -l)
+
+  # save config
+  local saved_config="null"
+  local enabled=false
+  local pod_cnt=0
+
+  if [ "$cnt" -gt 0 ]; then
+    # in v1.1.2, the monitoring is enabled
+    enabled=true
+    saved_config=$(kubectl get managedchart -n fleet-local "$chart_name" -o yaml | yq -e ".spec.values")
+    echo "saved managedchart config:"
+    echo "$saved_config"
+    pod_cnt=$(kubectl get pods -n "$namespace" --no-headers | wc -l)
+    echo "there are $pod_cnt pods in $namespace will be replaced"
+    echo "delete managedchart $chart_name"
+    kubectl delete managedchart -n fleet-local "$chart_name"
```

Comment:
If we delete the chart, will services with PVC still have previous data after the conversion?
Like Prometheus and alert manager.

Reply:
grafana is a deployment, while the prometheus and alertmanager are statefulsets

k8s doesnt gc pvc's associated with statefulsets.

Reply:
It explains the differences. thanks.